master/src/main/java/org/apache/celeborn/service/deploy/master/MasterSecretRegistryImpl.java

```diff
@@ -44,7 +44,32 @@ public void register(String appId, String secret) {
     }
   }
 
-  void setMetadataHandler(IMetadataHandler metadataHandler) {
+  @Override
+  public String getSecretKey(String appId) {
+    String secret = super.getSecretKey(appId);
+    if (secret == null && metadataHandler != null) {
+      LOG.info("Fetching secret from AbstractMetaManager for appId: {}", appId);
+      ApplicationMeta applicationMeta = metadataHandler.applicationMetas.get(appId);
+      if (applicationMeta != null) {
+        secret = applicationMeta.secret();
+      } else {
+        LOG.warn("Secret not found from AbstractMetaManager for appId: {}", appId);
+      }
+    }
+    return secret;
+  }
+
+  @Override
+  public boolean isRegistered(String appId) {
+    boolean isRegistered = super.isRegistered(appId);
+    if (!isRegistered && metadataHandler != null) {
+      LOG.info("Fetching registration status from AbstractMetaManager for appId: {}", appId);
+      isRegistered = metadataHandler.applicationMetas.containsKey(appId);
```

Comment:
nit: shall we also call `super.register` if isRegistered is true?

Reply:
sg, I added the super.register(appId, secret) to both `getSecretKey` and `isRegistered` if the secret is only in ratis.